ai2thor/controller.py

```diff
@@ -671,24 +737,44 @@ def step(self, action=None, **action_args):
         action = copy.deepcopy(action)
 
         # XXX should be able to get rid of this with some sort of deprecation warning
-        if 'AI2THOR_VISIBILITY_DISTANCE' in os.environ:
-            action['visibilityDistance'] = float(os.environ['AI2THOR_VISIBILITY_DISTANCE'])
+        if "AI2THOR_VISIBILITY_DISTANCE" in os.environ:
+            action["visibilityDistance"] = float(
+                os.environ["AI2THOR_VISIBILITY_DISTANCE"]
+            )
 
         self.last_action = action
 
-        rotation = action.get('rotation')
+        rotation = action.get("rotation")
         if rotation is not None and type(rotation) != dict:
-            action['rotation'] = {}
-            action['rotation']['y'] = rotation
+            action["rotation"] = {}
+            action["rotation"]["y"] = rotation
+
+        # support for deprecated parameter names (old: new)
+        changed_parameter_names = {
+            "renderClassImage": "renderSemanticSegmentation",
+            "renderObjectImage": "renderInstanceSegmentation",
+        }
+        for old, new in changed_parameter_names.items():
```

Comment:
this would then become:

```
for old, new in change_parameter_names.get(action['action'], {}).items():
```
or slightly more verbose but faster:
```
if action['action'] in changed_parameter_names:
    for old, new in change_parameter_names[action['action']].items():

```